controllers/runnermanager.go

```diff
@@ -126,21 +133,30 @@ func newManagerLoop(log logr.Logger, interval time.Duration, k8sClient client.Cl
 		repository:            rp.Spec.RepositoryName,
 		replicas:              rp.Spec.Replicas,
 		maxRunnerPods:         rp.Spec.MaxRunnerPods,
+		slackAgentClient:      agentClient,
 		slackChannel:          rp.Spec.SlackAgent.Channel,
 		slackAgentServiceName: rp.Spec.SlackAgent.ServiceName,
 		recreateDeadline:      recreateDeadline,
 		lastCheckTime:         time.Now().UTC(),
 	}
-	return loop
+	return loop, nil
 }
 
-func (m *managerLoop) update(rp *meowsv1alpha1.RunnerPool) {
+func (m *managerLoop) update(rp *meowsv1alpha1.RunnerPool) error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	m.replicas = rp.Spec.Replicas
 	m.maxRunnerPods = rp.Spec.MaxRunnerPods
 	m.slackChannel = rp.Spec.SlackAgent.Channel
+	if m.slackAgentServiceName == rp.Spec.SlackAgent.ServiceName {
+		return nil
+	}
+	err := m.slackAgentClient.UpdateServerURL(rp.Spec.SlackAgent.ServiceName)
+	if err != nil {
+		return err
+	}
 	m.slackAgentServiceName = rp.Spec.SlackAgent.ServiceName
```

Comment:
In this case, it is better to put the update process inside the `if` block.

```suggestion
	if m.slackAgentServiceName != rp.Spec.SlackAgent.ServiceName {
		err := m.slackAgentClient.UpdateServerURL(rp.Spec.SlackAgent.ServiceName)
		if err != nil {
			return err
		}
		m.slackAgentServiceName = rp.Spec.SlackAgent.ServiceName
	}
```

Reply:
I have reflected it.